core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -54,7 +54,12 @@ class TaskInfo(
    * accumulable to be updated multiple times in a single task or for two accumulables with the
    * same name but different IDs to exist in a task.
    */
-  val accumulables = ListBuffer[AccumulableInfo]()
+  lazy val accumulables: ListBuffer[AccumulableInfo] = ListBuffer(_accumulables: _*)
```

Comment:
Isn't a `def` here the right thing? Otherwise, if `_accumulables` changes after this lazy val is evaluated, things will get out of sync.

Or given how the semantics here now are getting pretty weird, maybe changing it to an immutable list and just breaking compatibility would be better. Then you can just return `_accumulables`.